### PR TITLE
Merge SimulatorInstaller commands to main CLI

### PR DESCRIPTION
**Pull request by premun**
This removes the SimulatorInstaller project and moves the file to CLI so that you can do:
```
xharness apple simulators list
```

When installing the simulators, it doesn't accept the same format e.g. `ios-simulator-64` of targets as the CLI, this will be a follow-up.

#478

### CODE REVIEW

File: src/Microsoft.DotNet.XHarness.CLI/Program.cs [hunk 1]
```diff
@@ -28,27 +28,18 @@ public static int Main(string[] args)
...
 
-            // Root command: will use the platform specific commands to perform the appropriate action.
-            var commands = new CommandSet("xharness");
-
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
-            {
-                commands.Add(new AppleCommandSet());
-            }
-            else if (args[0] == "apple")
+            if (args[0] == "apple" && RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
```
Comment (mattleibow) [inline src/Microsoft.DotNet.XHarness.CLI/Program.cs:31]: This throws when it is just `dotnet xharness`:

```suggestion
            if (args.Length > 0 && args[0] == "apple" && !RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
```

File: src/Microsoft.DotNet.XHarness.CLI/Commands/XHarnessHelpCommand.cs [hunk 2]
```diff
@@ -20,6 +21,13 @@ public override int Invoke(IEnumerable<string> arguments)
         {
             string[] args = arguments.ToArray();
 
+            // TODO (#400): We can remove this after some time when users get used to the new commands
+            if (args[0] == "ios")
```
Comment (mattleibow) [inline src/Microsoft.DotNet.XHarness.CLI/Commands/XHarnessHelpCommand.cs:25]:
```suggestion
            if (args.Length > 0 && args[0] == "ios")
```

File: src/Microsoft.DotNet.XHarness.CLI/Program.cs [hunk 3]
```diff
@@ -28,27 +28,18 @@ public static int Main(string[] args)
                 args[0] = "apple";
```
Comment (mattleibow) [inline src/Microsoft.DotNet.XHarness.CLI/Program.cs:28]: The line above this also needs to be a Length > 0 as `dotnet ios` will fail without a warning as to the change.

**Review by mattleibow — COMMENTED** [review]
There are some argument testing that seems a little off, but otherwise seems to be working.

Thanks for this as we have been using all sorts of scripts and things to avoid having to add more unrelated code to repos. This is really going to help keep things clean.

**Comment by premun** [comment]
@greenEkatherine we would need this PR in because it changes names of PR legs and the branch policies had to change.

Can you please have a look at this PR?

**Review by greenEkatherine — APPROVED** [review]

**Pull request merged**